perf_test/sparse/KokkosSparse_gs.cpp

```diff
@@ -187,92 +188,170 @@ void runGS(const GS_Parameters& params) {
-          true, true, 1.0, params.sweeps);
-      break;
+
+  {
+    namespace KE = Kokkos::Experimental;
+    auto ns      = params.nstreams;
+    auto es      = exec_space();
+    if (ns == 1)
+      instances = KE::partition_space(es, 1);
+    else if (ns == 2)
+      instances = KE::partition_space(es, 1, 1);
+    else if (ns == 3)
+      instances = KE::partition_space(es, 1, 1, 1);
+    else if (ns == 4)
+      instances = KE::partition_space(es, 1, 1, 1, 1);
+    else if (ns == 5)
+      instances = KE::partition_space(es, 1, 1, 1, 1, 1);
+    else if (ns == 6)
+      instances = KE::partition_space(es, 1, 1, 1, 1, 1, 1);
+    else if (ns == 7)
+      instances = KE::partition_space(es, 1, 1, 1, 1, 1, 1, 1);
+    else if (ns == 8)
+      instances = KE::partition_space(es, 1, 1, 1, 1, 1, 1, 1, 1);
+    else
+      Kokkos::abort("--streams outside of [1, 8] is not supported.");
   }
 
-  double applyLaunchTime = timer.seconds();
+  double blockExtractionTime = 0, symbolicLaunchTimeTotal = 0,
+         symbolicComputeTimeTotal = 0, numericLaunchTimeTotal = 0,
+         numericComputeTimeTotal = 0, applyLaunchTimeTotal = 0,
+         applyComputeTimeTotal = 0;
+
   timer.reset();
+  KokkosSparse::Impl::kk_extract_diagonal_blocks_crsmatrix_sequential(A,
+                                                                      DiagBlks);
   Kokkos::fence();
-  double applyComputeTime = timer.seconds();
-  timer.reset();
-  kh.destroy_gs_handle();
-  // Now, compute the 2-norm of residual
-  scalar_view_t res("Ax-b", nrows);
-  Kokkos::deep_copy(res, b);
-  scalar_t alpha = Kokkos::reduction_identity<scalar_t>::prod();
-  scalar_t beta  = -alpha;
-  KokkosSparse::spmv<scalar_t, crsMat_t, scalar_view_t, scalar_t,
-                     scalar_view_t>("N", alpha, A, x, beta, res);
-  double resnorm = KokkosBlas::nrm2(res);
-  std::cout << "\n*** Symbolic launch time: " << symbolicLaunchTime << '\n';
-  std::cout << "\n*** Symbolic compute time: " << symbolicComputeTime << '\n';
-  std::cout << "\n*** Numeric launch time: " << numericLaunchTime << '\n';
-  std::cout << "\n*** Numeric compute time: " << numericComputeTime << '\n';
-  std::cout << "\n*** Apply launch time: " << applyLaunchTime << '\n';
-  std::cout << "\n*** Apply compute time: " << applyComputeTime << '\n';
-  // note: this still works if the solution diverges
-  std::cout << "Relative res norm: " << resnorm / bnorm << '\n';
+  blockExtractionTime = timer.seconds();
+
+  for (int i = 0; i < params.nstreams; i++) {
+    auto blk_A     = DiagBlks[i];
+    auto blk_nrows = blk_A.numRows();
+    auto blk_ncols = blk_A.numCols();
+    if (blk_nrows != blk_ncols) {
+      cout << "ERROR: Gauss-Seidel only works for square matrices\n";
+      Kokkos::finalize();
+      exit(1);
+    }
+    b[i] = scalar_view_t("b[" + std::to_string(i) + "]", blk_nrows);
+    x[i] = scalar_view_t("x[" + std::to_string(i) + "]", blk_nrows);
+    {
+      srand(54321);
```

Comment:
```suggestion
      srand(54321 + i);
```

Wouldn't hurt to have different RHS's for each block